distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/config/UrlResourceReaderConfigurator.java

```diff
@@ -30,28 +31,63 @@
 
   private static final String PID = "ddf.catalog.resource.impl.URLResourceReader";
 
+  private static final String ROOT_RESOURCE_DIRECTORIES_PROPERTY_KEY = "rootResourceDirectories";
+
   private ConfigurationAdmin configAdmin;
 
   public UrlResourceReaderConfigurator(ConfigurationAdmin configAdmin) {
     this.configAdmin = configAdmin;
   }
 
   public void setUrlResourceReaderRootDirs(String... rootResourceDirs) throws IOException {
+    Set<String> newRootResourceDirs = ImmutableSet.<String>builder().add(rootResourceDirs).build();
+
     Configuration configuration = configAdmin.getConfiguration(PID, null);
+
+    if (configuration == null) {
+      LOGGER.warn(
+          "{} configuration was null, cannot update {}",
+          PID,
+          ROOT_RESOURCE_DIRECTORIES_PROPERTY_KEY);
+      return;
+    }
+
+    if (configuration.getProperties() != null) {
```

Comment:
✏️ Should avoid calling `getProperties()` twice. Depending on the implementation, it might not actually be a simple getter. It happens to be the case here.